src/main/java/org/apache/maven/shared/jar/classes/JarRuntimeVersion.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.maven.shared.jar.classes;
+
+import java.util.List;
+import java.util.jar.JarEntry;
+
+public class JarRuntimeVersion {
+
+    /**
+     * The JAR entries.
+     */
+    private final List<JarEntry> entries;
+
+    private final JarClasses jarClasses;
+
+    public JarRuntimeVersion(List<JarEntry> entries, JarClasses jarClasses) {
```

Comment:
I have the feeling that the actual versio should be part of this class and not mapped through a `Map`. We have `stream()` to filter for that or the holder can maintain an internal map for fast access...

Reply:
The version could be easily added, but an instance of JarRuntimeVersions is one of such holders.

Reply:
Right, we we have the in the level about, leave as-is.